src/ashell/shell-state.c

```diff
@@ -594,8 +594,10 @@ int32_t ashell_set_bootcfg(char *buf)
         return RET_ERROR;
     }
 
-    ssize_t written = fs_write(file, buf, strlen(buf));
-    if (written == 0) {
+    ssize_t written = fs_write(file, __TIMESTAMP__, strlen(__TIMESTAMP__));
```

Comment:
I wasn't too clear: w/ __TIMESTAMP__, I think if you don't touch shell-state.c but rebuild ashell to get other changes, the timestamp won't actually change and you won't remove the boot.cfg.